src/hooks/useResizer.js

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+
+import { addActions, actions } from "../actions";
+import { defaultState } from "./useTableState";
+
+defaultState.resizedColumns = {};
+
+addActions({
+  resizeColumn: "__resizeColumn__"
+});
+
+const propTypes = {
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      width: PropTypes.number
+    })
+  ),
+  onResizedChange: PropTypes.func
+};
+
+export const useResizer = props => {
+  PropTypes.checkPropTypes(propTypes, props, "property", "useResizer");
+
+  const {
+    columns,
+    hooks: { headers: headerHooks },
+    state: [{ resizedColumns }, setState],
+    getResizerProps,
+    onResizedChange
+  } = props;
+  const currentlyResizingInfo = useRef(undefined);
+
+  useEffect(() => {
+    if (resizedColumns) {
+      Object.keys(resizedColumns).forEach(index => (columns[index].width = resizedColumns[index]));
+    }
```

Comment:
I'm confused by what this is doing. Is this updating the column with the new width? Seems like this should be done in a hook somewhere, no?

Reply:
This is how i'm triggering a re-render when the columns widths change.  

If you were to put it somewhere else, where were you thinking and how would it be done?